### PR TITLE
Add macros for the KDF algorithm names.

### PR DESCRIPTION
**Pull request by paulidale**
This avoids the problems with PBKDF2 and SCRYPT not being of the same form
as the rest.

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: include/openssl/core_names.h [hunk 1]
```diff
@@ -128,6 +130,16 @@ extern "C" {
...
 
+/* Known KDF names */
+#define OSSL_KDF_NAME_HKDF          SN_hkdf
+#define OSSL_KDF_NAME_PBKDF2        LN_id_pbkdf2
+#define OSSL_KDF_NAME_SCRYPT        SN_id_scrypt
+#define OSSL_KDF_NAME_SSHKDF        SN_sshkdf
+#define OSSL_KDF_NAME_SSKDF         SN_sskdf
+#define OSSL_KDF_NAME_TLS1_PRF      SN_tls1_prf
+#define OSSL_KDF_NAME_X942KDF       SN_x942kdf
+#define OSSL_KDF_NAME_X963KDF       SN_x963kdf
```
Comment (levitte) [inline include/openssl/core_names.h]: I would rather have the explicit strings here

Comment (paulidale) [reply include/openssl/core_names.h]: Changed.

I'm not keen on introducing unnecessary redundancy but not sufficiently against it to bother.

**Comment by paulidale** [comment]
Ping?

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**